core/src/main/scala/kafka/coordinator/GroupMetadataManager.scala

```diff
@@ -659,10 +660,10 @@ class GroupMetadataManager(val brokerId: Int,
       config.offsetsTopicNumPartitions
   }
 
-  private def getMessageFormatVersionAndTimestamp(partition: Int): (Byte, Long) = {
+  private def getMessageFormatVersionAndTimestampIfReplicaIsLocal(partition: Int): (Byte, Long) = {
```

Comment:
The other advantage of returning `Option` is that it will force us to handle other cases that could have a similar issue.